src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -0,0 +1,72 @@
+from snuba_sdk import Column, Function
+
+from sentry.sentry_metrics.utils import resolve_weak
+
+
+def __counter_sum_aggregation_on_session_status_factory(session_status, metric_ids, alias=None):
+    return Function(
+        "sumMergeIf",
+        [
+            Column("value"),
+            Function(
+                "and",
+                [
+                    Function(
+                        "equals",
+                        [
+                            Column(f"tags[{resolve_weak('session.status')}]"),
+                            resolve_weak(session_status),
+                        ],
+                    ),
+                    Function("in", [Column("metric_id"), list(metric_ids)]),
+                ],
+            ),
```

Comment:
Note: I think you could rewrite this as (not saying it's necessarily better though) :

```suggestion
            And(
                [
                    Condition(
                         Column(f"tags[{resolve_weak('session.status')}]"),
                         Op.EQ,
                         resolve_weak(session_status),
                    ),
                    Condition(Column("metric_id"), Op.IN, list(metric_ids))),
                ],
            ),
```

Reply:
hmm what is the advantage of renaming this to `And` then?

Reply:
Reached out to @evanh on this.. Lets see which is preferable.. In any case, I will merge this and put up a followup PR if its needed to make this change

Reply:
In theory they are functionally equivalent. The `And` operator is syntactic sugar for `Function("and"`. However this won't work since SnQL doesn't support infix conditions outside of the where clause, and I suspect (not sure offhand) that the validation would fail since Function don't support `BooleanCondition` as a parameter.

Reply:
Thanks @evanh

Reply:
> However this won't work since SnQL doesn't support infix conditions outside of the where clause

Good to know, thanks!